### PR TITLE
add nnet3 tdnn recipe on swbd and its results

### PR DESCRIPTION
**Pull request by freewym**
Hi Vijay,

After pull the latest ctc branch, I closed the previous pull request and create a new one.

Yiming

### CODE REVIEW

File: egs/swbd/s5c/local/nnet3/run_tdnn.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+local/nnet3/run_ivector_common.sh --stage $stage || exit 1;
+if [ $stage -le 8 ]; then
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $dir/egs/storage ]; then
+    utils/create_split_dir.pl \
+     /export/b0{3,4,5,6}/$USER/kaldi-data/egs/wsj-$(date +'%m_%d_%H_%M')/s5/$dir/egs/storage $dir/egs/storage
+  fi
+
+  steps/nnet3/train_tdnn.sh --stage $train_stage \
+    --num-epochs 2 --num-jobs-initial 3 --num-jobs-final 16 \
+    --splice-indexes "-2,-1,0,1,2 -1,0,1,2 -3,-2,-1,0,1,2,3 -7,-6,-5,-4,-3,-2,-1,0,1,2" \
```
Comment (vijayaditya) [inline egs/swbd/s5c/local/nnet3/run_tdnn.sh]: Please use 
--splice-indexes="-2,-1,0,1,2  -1,2  -3,3  -7,2 0"

File: egs/swbd/s5c/local/nnet3/run_tdnn.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    --feat-type raw \
+    --online-ivector-dir exp/nnet3/ivectors_${train_set} \
+    --cmvn-opts "--norm-means=false --norm-vars=false" \
+    --initial-effective-lrate 0.0017 --final-effective-lrate 0.00017 \
+    --cmd "$decode_cmd" \
+    --pnorm-input-dim 2750 \
```
Comment (vijayaditya) [inline egs/swbd/s5c/local/nnet3/run_tdnn.sh]: Please use relu units in place of the pnorm units. 
I would suggest 1024 units.

File: egs/swbd/s5c/local/nnet3/run_tdnn.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+local/nnet3/run_ivector_common.sh --stage $stage || exit 1;
```
Comment (vijayaditya) [inline egs/swbd/s5c/local/nnet3/run_tdnn.sh]: Please pass on the option for speed perturbation to the run_ivector_common.sh script. You could do something like 

<pre><code>
suffix=
if [ "$speed_perturb" == "true" ]; then
  suffix=_sp
fi
dir=exp/nnet3/tdnn
dir=$dir${affix:+_$affix}
dir=${dir}$suffix
train_set=train_nodup$suffix
ali_dir=exp/tri4_ali_nodup$suffix

local/nnet3/run_ivector_common.sh --stage $stage \
  --speed-perturb $speed_perturb || exit 1;
</code></pre>

**Comment by freewym** [comment]
Thanks. I will rerun the script and update the results.

Yiming

**Comment by freewym** [comment]
The updated script and results are pushed in.

Yiming

File: egs/swbd/s5c/local/nnet3/run_tdnn.sh [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+where "nvcc" is installed.
+EOF
+fi
+
+suffix=
+if [ "$speed_perturb" == "true" ]; then
+  suffix=_sp
+fi
+dir=exp/nnet3/tdnn
+dir=$dir${affix:+_$affix}${use_delay:+_ld$label_delay}
```
Comment (vijayaditya) [inline egs/swbd/s5c/local/nnet3/run_tdnn.sh]: there is no label delay used in tdnn recipe.

File: egs/swbd/s5c/local/nnet3/run_tdnn.sh [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+dir=${dir}$suffix
+train_set=train_nodup$suffix
+ali_dir=exp/tri4_ali_nodup$suffix
+
+local/nnet3/run_ivector_common.sh --stage $stage \
+	--speed-perturb $speed_perturb || exit 1;
+if [ $stage -le 8 ]; then
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $dir/egs/storage ]; then
+    utils/create_split_dir.pl \
+     /export/b0{3,4,5,6}/$USER/kaldi-data/egs/wsj-$(date +'%m_%d_%H_%M')/s5/$dir/egs/storage $dir/egs/storage
```
Comment (vijayaditya) [inline egs/swbd/s5c/local/nnet3/run_tdnn.sh]: its 
egs/swbd not egs/wsj

Comment (freewym) [inline egs/swbd/s5c/local/nnet3/run_tdnn.sh]: Fixed

File: egs/swbd/s5c/local/nnet3/run_tdnn.sh [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+fi
+dir=exp/nnet3/tdnn
+dir=$dir${affix:+_$affix}
+dir=${dir}$suffix
+train_set=train_nodup$suffix
+ali_dir=exp/tri4_ali_nodup$suffix
+
+local/nnet3/run_ivector_common.sh --stage $stage \
+	--speed-perturb $speed_perturb || exit 1;
+if [ $stage -le 8 ]; then
```
Comment (vijayaditya) [inline egs/swbd/s5c/local/nnet3/run_tdnn.sh]: this has to be stage 9

Comment (freewym) [inline egs/swbd/s5c/local/nnet3/run_tdnn.sh]: Done.

**Comment by vijayaditya** [comment]
@danpovey good to merge

**Comment by danpovey** [comment]
Thanks!  Merging.

**Comment by danpovey** [comment]
Vijay, since this was in the ctc branch, you might want to merge this along with the other changes.

**Comment by naxingyu** [comment]
I will delete the old nnet3/lstm results.
One more thing, I checked run_ivector_common.sh, but didn't find where 
'volume perturb' happens.
At line-102, the comment says "do volume perturbation", but the 
following make_mfcc.sh is extracting hires mfcc on speed perturbed data.
What did I miss?

Xingyu

**Comment by vijayaditya** [comment]
@naxingyu thanks for pointing out. This is a bug. I should have created the _sp directories from train_hires directory which was volume perturbed. Will push the changes soon.

**Pull request merged**